### PR TITLE
Mes 3668 health declaration cat be refactor

### PR DESCRIPTION
**Pull request by faggc**
## Description
Addition of BE page for Health Declaration

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/health-declaration/cat-be/health-declaration.cat-be.page.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<ion-header>
+  <practice-mode-banner *ngIf="isPracticeMode"></practice-mode-banner>
```
Comment (MarkPatt89) [inline src/pages/health-declaration/cat-be/health-declaration.cat-be.page.html]: You can remove this as we don't have a practice mode for B+E

File: src/pages/health-declaration/cat-be/health-declaration.cat-be.page.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+  passCertificateNumber$: Observable<string>;
+  licenseProvided$: Observable<boolean>;
+  conductedLanguage$: Observable<string>;
+}
+@IonicPage()
+@Component({
+  selector: 'health-declaration-cat-be-page',
+  templateUrl: 'health-declaration.cat-be.page.html',
+})
+export class HealthDeclarationCatBEPage extends PracticeableBasePageComponent {
```
Comment (MarkPatt89) [inline src/pages/health-declaration/cat-be/health-declaration.cat-be.page.ts]: Again this should extend `BasePageComponent` as we don't use practice mode.

File: src/pages/health-declaration/cat-be/health-declaration.cat-be.page.html (hunk 1 above)
Comment (appleabs) [inline src/pages/health-declaration/cat-be/health-declaration.cat-be.page.html]: Category BE doesn't need practice mode

File: src/pages/health-declaration/cat-be/health-declaration.cat-be.page.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+  }
+
+  persistAndNavigate(resetLicenseProvided: boolean) {
+    this.deviceAuthenticationProvider.triggerLockScreen()
+      .then(() => {
+        if (this.licenseProvided && resetLicenseProvided) {
+          this.store$.dispatch(new ProvisionalLicenseNotReceived());
+        }
+        this.store$.dispatch(new ContinueFromDeclaration());
+        this.navController.push(CAT_B.BACK_TO_OFFICE_PAGE).then(() => {
```
Comment (MarkPatt89) [inline src/pages/health-declaration/cat-be/health-declaration.cat-be.page.ts]: Should navigate to `CAT_BE` page

File: src/pages/health-declaration/cat-be/health-declaration.cat-be.page.ts (hunk 2 above)
Comment (appleabs) [inline src/pages/health-declaration/cat-be/health-declaration.cat-be.page.ts]: No practice mode needed so would extend base page component rather that practicibleBasePageComponent

**Review by appleabs — APPROVED** [review]
nice work :)

**Review by MarkPatt89 — APPROVED** [review]
Nice work!

**Review by ammarhaiderbjss — APPROVED** [review]

**Pull request merged**